File: CPP Unit 06/Static/Q1.cpp
Single File Programming Question
Problem Statement



Hawkeye is practicing his archery skills at the Avengers training ground. He needs to calculate the flight time for multiple arrows using the formula: Time = Distance / Velocity. To store and display the flight times of all his arrows, he uses a list data structure.



Help him to implement the task using the list container from the Standard Template Library (STL).

Input format :
The first line of input consists of an integer n representing the number of arrows.

The next n pairs of lines consist of a double distance representing the distance to target in meters, followed by a double velocity representing the arrow velocity in meters per second.

Output format :
The first line of output prints "Flight times:".

The subsequent lines print the flight time for each arrow with two decimal places, one time per line.



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

1 ≤ n ≤ 100

0.0 < distance ≤ 1000.0

0.0 < velocity ≤ 500.0

Sample test cases :
Input 1 :
2
10.0
2.0
20.0
4.0
Output 1 :
Flight times:
5.00
5.00
Input 2 :
3
5.0
1.0
8.0
2.0
12.0
3.0
Output 2 :
Flight times:
5.00
4.00
4.00




#include <iostream>
#include <list>
#include <iomanip>
using namespace std;

int main() {
    int n;
    cin >> n;

    list<double> flightTimes;

    for (int i = 0; i < n; i++) {
        double distance, velocity;
        cin >> distance >> velocity;
        flightTimes.push_back(distance / velocity);
    }

    cout << "Flight times:" << endl;
    cout << fixed << setprecision(2);

    for (double t : flightTimes) {
        cout << t << endl;
    }

    return 0;
}

File: Java/Unit 01/Lec 08/Q1.java
Single File Programming Question
Problem Statement



Imagine you are building a discount calculator for a retail store. The program takes the original price and a discount type ('F' for flat discount or 'P' for percentage discount) as input. It then calculates and displays the discount amount and the final price.



Formulas Used:

For flat discount, the calculation is final price = original price - flat discount. 

For percentage discount, it is final price = original price - (original price * discount percentage / 100).

Input format :
The first line of input consists of a double value representing the original price.

The second line of input consists of a character representing the discount type ('F' for flat discount or 'P' for percentage discount).

The third line of input consists of a double value representing the discount amount (flat discount if the type is 'F', or discount percentage if the type is 'P').

Output format :
If the discount type is 'F':

The first line of output prints "Discount Amount: Rs." followed by the discount amount rounded to two decimal places.
The second line of output prints "Final Price: Rs." followed by the final price rounded to two decimal places.
If the discount type is 'P':

The first line of output prints "Discount Percentage: " followed by the discount percentage rounded to two decimal places.
The second line of output prints "Final Price: Rs." followed by the final price rounded to two decimal places.
If the discount type is invalid:

The output prints "Invalid discount type".


Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

1.0 ≤ price ≤ 1000.0

0.0 ≤ flat discount ≤ original price

0.0 ≤ discount percentage ≤ 200.0

Sample test cases :
Input 1 :
120.7
F
30.0
Output 1 :
Discount Amount: Rs.30.00
Final Price: Rs.90.70
Input 2 :
75.2
P
20.5
Output 2 :
Discount Percentage: 20.50
Final Price: Rs.59.78
Input 3 :
50.0
H
100.0
Output 3 :
Invalid discount type






import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        double price = sc.nextDouble();
        char type = sc.next().charAt(0);
        double value = sc.nextDouble();

        if (type == 'F') {
            double finalPrice = price - value;
            System.out.printf("Discount Amount: Rs.%.2f\n", value);
            System.out.printf("Final Price: Rs.%.2f\n", finalPrice);
        } 
        else if (type == 'P') {
            double discount = price * value / 100;
            double finalPrice = price - discount;
            System.out.printf("Discount Percentage: %.2f\n", value);
            System.out.printf("Final Price: Rs.%.2f\n", finalPrice);
        } 
        else {
            System.out.print("Invalid discount type");
        }

        sc.close();
    }
}






  
